.github/actions/calculate-next-version/README.md

```diff
@@ -0,0 +1,105 @@
+# Calculate Next Version
```

Comment:
I'm attaching this here but, it's really about the whole action here: this seems like a lot of code for bumping the minor or patch version. I suspect we've abstracted a lot and tried to handle a bunch of extra cases we don't need to (i.e. I suspect the completeness of this is YAGNI compared to the relatively smaller chunk of code doing the same thing in the branch: https://github.com/posit-dev/publisher/pull/3002/files#diff-443ddd6d357ddc79538a1cbaf01097a05bd139a4e7468c28f21f972cd091605cR72-R123)

Can we cut this down to the bare minimum we need to get automatic pre-releases out today and then we can work on a full version later if that is helpful for us? (which might be never, and that's ok!)